sched/scheduler/stateful_scheduler.go

```diff
@@ -107,6 +107,23 @@ func NewStatefulScheduler(
 	stat stats.StatsReceiver,
 ) *statefulScheduler {
 
+	nodeReadyFn := func(node cluster.Node) bool {
+		run := rf(node)
+		st, err := run.StatusAll()
+		if err != nil {
+			return false
+		}
+		if len(st) == 1 && st[0].Error == runner.NoRunnersMsg {
+			//TODO(jschiller) amend protocol to be more direct than this array of one with a particular error status.
```

Comment:
agreed - recommend adding a healthCheck() or Status() API - returning  starting, running, idle, or error (not necessarily in this pr)